Makefile

```diff
@@ -1,33 +1,27 @@
 PATH := $(eTCpath)$(PATH)
 
-.PHONY: ENIGMA all clean Game clean-game .FORCE
+.PHONY: ENIGMA all clean Game clean-game liblodepng libProtocols libGMX required-directories .FORCE
 
 ENIGMA: .FORCE libProtocols liblodepng
 	$(MAKE) -C CompilerSource 
 
-clean:
+clean: .FORCE
 	$(MAKE) -C CompilerSource/ clean
 	$(MAKE) -C CommandLine/emake/ clean
 	$(MAKE) -C CommandLine/libGMX/ clean
 	$(MAKE) -C CommandLine/protos/ clean
 	$(MAKE) -C CommandLine/testing/ clean
 	$(MAKE) -C lodepng/ clean
-	
-all:
-	$(MAKE) -C lodepng/
-	$(MAKE) -C CommandLine/protos/
-	$(MAKE) -C CommandLine/libGMX/
-	$(MAKE) -C CompilerSource/
-	$(MAKE) -C CommandLine/emake/
-	$(MAKE) -C CommandLine/testing/
 
-Game: liblodepng
+all: liblodepng libProtocols libGMX ENIGMA emake test-runner .FORCE
```

Comment:
Let me explain, a little. The `.FORCE` target is a dummy target I created that is never satisfied. It causes `make` to run the rules of a target, even when that target exists, because that target will always appear up-to-date unless you read its actual makefile. So that's what it's forcing to happen.

The `.PHONY` rule is a special rule that tells make that the file doesn't actually exist.

Reply:
whatever, did I use it incorrectly or someshit? if so tell me what to change

Reply:
or better yet, fix it yourself

Reply:
I believe he is saying that `.FORCE` is not needed with a `.PHONY` target because it's already going to run anyway. `clean` does need to be forced because it will never exist either. `all` should also need to be forced for the same reason as `clean`.